### DISCUSSION TITLE
Portrait textbox scaling

### DISCUSSION THREAD

**Discussion by VoltitanDev** [dialogic-godot/dialogic#2242] [Q&A] [answered]
I noticed that there  is a portrait textbox option in dialogic 2 but I was wondering if there is a way to scale the portrait within the textbox so that it only shows the head and shoulders of the character speaking rather than show the sprites  whole body like the original sprite that's displayed in the timeline.

**Comment by Jowan-Spooner** [comment]
Hey @VoltitanDev, games like that have not been done often with dialogic, so this is an underdeveloped area of dialogic.
There are basically two ways you could go about this. Whatever way, you will have to modify the layout to achieve them, so you'll have to make a custom version of the SpeakerTextbox layer.

A) You could actually resize the portrait container and mask it with a parent node (In the SpeakerTextbox this kind of masking is already setup, so you just have to make the portrait container extend to the bottom). This means you could use full-body portraits but essentially only display the upper part of it:
![grafik](https://github.com/dialogic-godot/dialogic/assets/42868150/02a1147d-717a-4888-ac73-1327ff0c6986)

B) Portrait containers that use the speaker mode have a setting called portrait-prefix (in the speaker section). This prefix can be used to use a different portrait for the speaker container.
For example if these are the portraits on your character:
```
- Happy
- Sad
- Curious
- Head (Group/Folder)
      - Happy
      - Sad
      - Curious
```
Then you could set the portrait-prefix to "Head/" on the speaker portrait container. If you characters portrait is set to "Sad" now, while the normal container will just use the Sad portrait, the speaker container will try to use "Head/Sad" instead. This way you can have custom pictures in the speaker textbox.

Truth be told, I've never actually used this, nor do I think anyone besides me knows this exists, as it's not mentioned in the documentation yet. Let me know if it works, in case you try it out.

**Reply by VoltitanDev** [reply]
For the custom Speakerbox, how does one access it through a scene?

**Reply by Jowan-Spooner** [reply, marked as answer]
In the styles editor add a "Speaker Textbox" layer and then use the make custom option. Or make a textbox from scratch. Checkout the Styles and Layout documentation: https://docs.dialogic.pro/styles-and-layouts.html There is also a video there, showcasing the workflow.
